### PR TITLE
Automatically close PRs against staging repos

### PR DESCRIPTION
**Pull request by cblecker**
This adds a new commenter job to close PRs against staging repos

fixes https://github.com/kubernetes/kubernetes/issues/122238

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
for review and discussion

File: config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml [hunk 1]
```diff
@@ -131,6 +131,84 @@ periodics:
...
+        Kubernetes Contributors:
+        If this PR is to update the `README.md` itself, please reopen this PR by commenting:
+        ```
+        /triage accepted
+        /reopen
+        ```
+
+        Please send feedback to sig-contributor-experience at [kubernetes/community](https://github.com/kubernetes/community).
+
+        /close not-planned
```
Comment (MadhavJivrajani) [inline config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:200]: We can just use `/close` here, there is no special `not_planned` status for PRs unfortunately.
xref: https://github.com/kubernetes/test-infra/pull/27545

File: config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml [hunk 2]
```diff
@@ -131,6 +131,84 @@ periodics:
...
+        If you're looking at contributing a code fix or feature, please review:
+        - The [Contributor Guide](https://git.k8s.io/community/contributors/guide)
+        - The [staging area README](https://git.k8s.io/kubernetes/staging#readme), which describes how the staging repos function and how to contribute
+
+        Kubernetes Contributors:
+        If this PR is to update the `README.md` itself, please reopen this PR by commenting:
+        ```
+        /triage accepted
+        /reopen
+        ```
```
Comment (MadhavJivrajani) [inline config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:196]: I'm likely missing something - can you pls point me to where we document that a PR to update the README is allowed for staging repos?

File: config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml [hunk 3]
```diff
@@ -131,6 +131,84 @@ periodics:
...
+        --comment=The Kubernetes project doesn't accept direct changes directly against this repository unless the change is to the `README.md` in the root of the repo.
+
+        If you're looking at contributing a code fix or feature, please review:
+        - The [Contributor Guide](https://git.k8s.io/community/contributors/guide)
+        - The [staging area README](https://git.k8s.io/kubernetes/staging#readme), which describes how the staging repos function and how to contribute
+
+        Kubernetes Contributors:
+        If this PR is to update the `README.md` itself, please reopen this PR by commenting:
+        ```
+        /triage accepted
```
Comment (MadhavJivrajani) [inline config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:194]: A consideration to keep in mind here would be `/triage accepted` can only be applied by org members iirc, if the contributor isn't an org member, they won't be able to re-open the PR.

**Review by MadhavJivrajani — COMMENTED** [review]
Thanks for starting this!
Leaving a few initial comments <3

File: config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml (hunk 2 above)
Comment (cblecker) [reply config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml]: It's permitted in client-go, but not the others

**Comment by cblecker** [comment]
I dug a bit more and found https://github.com/kubernetes/test-infra/pull/22879 -- we already configured a blockade but we never turned the plugin on in client-go, so this change never took effect.

I've updated the blockage configuration to account for the fact that you can only change README.md in client-go, but not the other repos. I've also updated the closer job to key off of the blockade label.

File: config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml [hunk 4]
```diff
@@ -131,6 +131,77 @@ periodics:
...
+        repo:kubernetes/metrics
+        repo:kubernetes/mount-utils
+        repo:kubernetes/pod-security-admission
+        repo:kubernetes/sample-apiserver
+        repo:kubernetes/sample-cli-plugin
+        repo:kubernetes/sample-controller
+      - --token=/etc/github-token/token
+      - --endpoint=http://ghproxy.default.svc.cluster.local
+      - |-
+        --comment=The Kubernetes project doesn't accept direct changes directly against this repository as outlined above.
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:185]: It feels kinda weird to duplicate this text in the blockade comment AND the robo-/close comment : thinking :

Comment (jsoref) [reply config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:185]: "Closing per above"? (if you're confident the message will always be present)

**Comment by jsoref** [comment]
@cblecker will you continue pushing this forward?

**Pull request closed without merging**